### PR TITLE
Implementation of Eclipse bug #469623

### PR DESCRIPTION
**Pull request by iweiss**
This code adds the ability to get headers using CharSequence instead of String for consistency.

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=469623

File: src/main/java/io/vertx/core/http/impl/HttpClientRequestImpl.java [hunk 1]
```diff
@@ -395,6 +395,11 @@ public String getHeader(String headerName) {
           }
 
           @Override
+          public String getHeader(CharSequence headerName) {
+            return resp.getHeader(headerName.toString());
```
Comment (purplefox) [inline src/main/java/io/vertx/core/http/impl/HttpClientRequestImpl.java]: why toString() ?

File: src/main/java/io/vertx/core/http/impl/HttpClientResponseImpl.java [hunk 2]
```diff
@@ -102,6 +102,11 @@ public String getHeader(String headerName) {
   }
 
   @Override
+  public String getHeader(CharSequence headerName) {
+    return headers().get(headerName.toString());
```
Comment (purplefox) [inline src/main/java/io/vertx/core/http/impl/HttpClientResponseImpl.java]: why toString() ?

File: src/main/java/io/vertx/core/http/impl/HttpServerRequestImpl.java [hunk 3]
```diff
@@ -169,6 +169,11 @@ public String getHeader(String headerName) {
   }
 
   @Override
+  public String getHeader(CharSequence headerName) {
+    return headers().get(headerName.toString());
```
Comment (purplefox) [inline src/main/java/io/vertx/core/http/impl/HttpServerRequestImpl.java]: why toString()?

File: src/main/java/io/vertx/core/http/impl/HttpClientResponseImpl.java (hunk 2 above)
Comment (iweiss) [inline src/main/java/io/vertx/core/http/impl/HttpClientResponseImpl.java]: Sorry, force of habit.

File: src/main/java/io/vertx/core/http/impl/HttpServerRequestImpl.java (hunk 3 above)
Comment (iweiss) [inline src/main/java/io/vertx/core/http/impl/HttpServerRequestImpl.java]: Sorry, force of habit. I will refactor this and re-submit.

File: src/main/java/io/vertx/core/http/impl/HttpClientRequestImpl.java (hunk 1 above)
Comment (iweiss) [inline src/main/java/io/vertx/core/http/impl/HttpClientRequestImpl.java]: Sorry, force of habit. I will refactor this and re-submit.

**Comment by iweiss** [comment]
I think this is correct now

File: src/test/java/io/vertx/test/core/HttpTest.java [hunk 4]
```diff
@@ -1262,6 +1262,33 @@ private void testRequestHeaders(boolean individually) {
...
+  public void testRequestHeadersWithCharSequence() {
+    HashMap<CharSequence, String> headers = new HashMap<>();
+    headers.put(HttpHeaders.TEXT_HTML, "text/html");
+    headers.put(HttpHeaders.USER_AGENT, "User-Agent");
+    headers.put(HttpHeaders.APPLICATION_X_WWW_FORM_URLENCODED, "application/x-www-form-urlencoded");
+
+    server.requestHandler(req -> {
+      assertEquals(headers.size() + 1, req.headers().size());
+
+      headers.forEach((k, v) -> assertEquals(v, req.headers().get(k)));
```
Comment (purplefox) [inline src/test/java/io/vertx/test/core/HttpTest.java]: Is this really testing the new API? Or is it just calling the old methods that take a String?

Comment (iweiss) [inline src/test/java/io/vertx/test/core/HttpTest.java]: It is. The new methods get a CharSequence and return a String.

**Comment by purplefox** [comment]
Ok thanks.

Can you make sure the Eclipse CLA and have signed the commits? The cla checker is complaining.

**Comment by iweiss** [comment]
I will try and open a new PR since this seems to be tainted by my git mistakes. Eventually I got everything right but it's complaining about previous commits.

**Comment by purplefox** [comment]
ok, cool thanks

**Pull request closed without merging**